jupyter_desktop/__init__.py

```diff
@@ -1,20 +1,36 @@
+from contextlib import closing
 import os
+import socket
 import tempfile
 
-
 HERE = os.path.dirname(os.path.abspath(__file__))
 
+def available_port():
+    '''Linux returns a random port as low as 1024. We need something >= 5900
+       to make the display math work.'''
+    for port in range(5900, 65535):
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+            try:
+                s.bind(('', port))
+            except Exception:
+                continue
+            else:
+                s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+                break
+    return port
+
 def setup_desktop():
     # make a secure temporary directory for sockets
     # This is only readable, writeable & searchable by our uid
     sockets_dir = tempfile.mkdtemp()
     sockets_path = os.path.join(sockets_dir, 'vnc-socket')
+    rfbport = available_port()
```

Comment:
If there are multiple vncservers the socket needs to be unique, so perhaps add
`sockets_path = os.path.join(sockets_dir, 'vnc-socket.' + str(rfbport))`?